pkg/kates/accumulator.go

```diff
@@ -165,45 +173,57 @@ func newAccumulator(ctx context.Context, client *Client, queries ...Query) (*Acc
 		client.watchRaw(ctx, q, rawUpdateCh, client.cliFor(field.mapping, q.Namespace))
 	}
 
-	acc := &Accumulator{client, fields, map[string]bool{}, 0, changed, sync.Mutex{}}
-
-	// This coalesces reads from rawUpdateCh to notifications that changes are available to be
-	// processed. This loop along with the logic in storeField guarantees the 3
-	// Goals/Requirements listed in the documentation for the Accumulator struct, i.e. Ensuring
-	// all Kinds are bootstrapped before any notification occurs, as well as ensuring that we
-	// continue to coalesce updates in the background while business logic is executing in order
-	// to ensure graceful load shedding.
-	go func() {
-		canSend := false
-
-		for {
-			var rawUp rawUpdate
-			if canSend {
-				select {
-				case changed <- struct{}{}:
-					canSend = false
-					continue
-				case rawUp = <-rawUpdateCh:
-				case <-ctx.Done():
-					return
-				}
+	acc := &Accumulator{
+		client:   client,
+		fields:   fields,
+		excluded: map[string]bool{},
+		synced:   0,
+		changed:  changed,
+		mutex:    sync.Mutex{},
+	}
+
+	go acc.Listen(ctx, rawUpdateCh, client.maxAccumulatorInterval)
+
+	return acc, nil
+}
+
+// This coalesces reads from rawUpdateCh to notifications that changes are available to be
+// processed. This loop along with the logic in storeField guarantees the 3
+// Goals/Requirements listed in the documentation for the Accumulator struct, i.e. Ensuring
+// all Kinds are bootstrapped before any notification occurs, as well as ensuring that we
+// continue to coalesce updates in the background while business logic is executing in order
+// to ensure graceful load shedding.
+func (a *Accumulator) Listen(ctx context.Context, rawUpdateCh <-chan rawUpdate, interval time.Duration) {
+	ticker := time.NewTicker(interval)
+	defer ticker.Stop()
+	var changeStatus changeStatus
+	var lastChangeSent time.Time
+	var synced bool
+
+	sendUpdate := func() {
+		a.changed <- struct{}{}
+		changeStatus = dispatched
+		lastChangeSent = time.Now()
+	}
+
+	for {
+		select {
```

Comment:
Just for clarity sake.

If we set an interval of 30 seconds. We'll have two paths here 

- We get new data and it has been past our set interval since we last updated anything. We go ahead and immediately send that
- We get new data but we just recently set data without our interval limit. We'll now wait until we get our `Tick`

We should maybe put up a brief comment that highlights the intended behavior here?